src/Kathara/manager/kubernetes/KubernetesNamespace.py

```diff
@@ -80,6 +84,27 @@ def get_namespace(self, lab_hash: str) -> Optional[client.V1Namespace]:
         namespace = self.client.list_namespace(label_selector=f"kubernetes.io/metadata.name={lab_hash}").items
         return namespace.pop() if namespace else None
 
+
+    def _create_private_registry_secret(self, lab_hash: str, pr_dcj: str) -> None:
```

Comment:
I am not sure this is the right place to put this method.

The secret is not something related to namespaces, but rather a completely different entity.

I would suggest to create a new class to handle secrets: `KubernetesSecret`, so we decouple the secrets logic from the namespace (that could be used for other things in the future).

I expect at least two methods:
- `create`, which takes the lab hash, the name of the secret and the data dict
- `delete` that takes lab hash and the name of the secret

Reply:
Seems good. There is a finite number of built-in secret types and each of them can have different requirements.
> [... if you are using one of the built-in types, you must meet all the requirements defined for that type.](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types)

For this reason I think it is better to define separate functions for the secret creation. In my case I could define the function `def create_dockerconfigjson_secret` to create a secret of type `kubernetes.io/dockerconfigjson`.

The secret deletion does only require the secret name, so there is no need to create different functions.